ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -95,67 +96,70 @@ public Gas gasRequirement(final BytesValue input) {
 
   @Override
   public BytesValue compute(final BytesValue input, final MessageFrame messageFrame) {
+    final String key = new String(input.extractArray(), UTF_8);
+    final ReceiveRequest receiveRequest = new ReceiveRequest(key, enclavePublicKey);
+
+    ReceiveResponse receiveResponse;
     try {
-      String key = new String(input.extractArray(), UTF_8);
-      ReceiveRequest receiveRequest = new ReceiveRequest(key, enclavePublicKey);
-      ReceiveResponse receiveResponse = enclave.receive(receiveRequest);
-
-      final BytesValueRLPInput bytesValueRLPInput =
-          new BytesValueRLPInput(
-              BytesValue.wrap(Base64.getDecoder().decode(receiveResponse.getPayload())), false);
-
-      PrivateTransaction privateTransaction = PrivateTransaction.readFrom(bytesValueRLPInput);
-
-      WorldUpdater publicWorldState = messageFrame.getWorldState();
-
-      BytesValue privacyGroupId =
-          BytesValue.wrap(receiveResponse.getPrivacyGroupId().getBytes(Charsets.UTF_8));
-      // get the last world state root hash - or create a new one
-      Hash lastRootHash =
-          privateStateStorage.getPrivateAccountState(privacyGroupId).orElse(EMPTY_ROOT_HASH);
-      MutableWorldState disposablePrivateState =
-          privateWorldStateArchive.getMutable(lastRootHash).get();
-
-      WorldUpdater privateWorldStateUpdater = disposablePrivateState.updater();
-
-      TransactionProcessor.Result result =
-          privateTransactionProcessor.processTransaction(
-              messageFrame.getBlockchain(),
-              publicWorldState,
-              privateWorldStateUpdater,
-              messageFrame.getBlockHeader(),
-              privateTransaction,
-              messageFrame.getMiningBeneficiary(),
-              OperationTracer.NO_TRACING,
-              messageFrame.getBlockHashLookup(),
-              privacyGroupId);
-
-      if (result.isInvalid() || !result.isSuccessful()) {
-        throw new Exception("Unable to process the private transaction");
-      }
+      receiveResponse = enclave.receive(receiveRequest);
+    } catch (IOException e) {
+      LOG.debug("Enclave probably does not have private transaction with key {}.", key);
```

Comment:
Can we point the user at which key they need to get lined up?  where does it live?

Reply:
The key here is a key in the sense of key -> value mapping. It is generated by orion and is used to propagate the private transactions between other orion nodes. If you do not have a private transaction with that key they either you are not in the privacy group or you have not received it yet (which should never ever ever happen).

It is also possible that the orion node is down at this point.

I have added the error in the log since this will only log in debug mode.